File: 백준/Silver/4949. 균형잡힌 세상/균형잡힌 세상.js
const fs = require("fs")
const filePath = "/dev/stdin"
let input = fs.readFileSync(filePath).toString().split("\n")
for (string of input){
    if (string==='.'){
        break
    }
    let parenthesis = []
    let open_parenthesis = new Array('(', '[')
    for (i of string){
        if (open_parenthesis.includes(i)){
            parenthesis.push(i)
        } else if (i === ')'){
            try {
                if (parenthesis.pop() === '('){
                    continue
                } else {
                    console.log("no")
                    break
                }
            } catch {
                console.log("no")
                break
            }
        } else if (i === ']'){
            try {
                if (parenthesis.pop() === '['){
                    continue
                } else {
                    console.log("no")
                    break
                }
            } catch {
                console.log("no")
                break
            }
        } else if (i === '.'){
            if (parenthesis.length != 0){
                console.log("no")
            } else {
                console.log("yes")
            }
        }
    }
}

File: 백준/Silver/4949. 균형잡힌 세상/균형잡힌 세상.py
while True:
    string = input()
    if string == '.':
        break
    parenthesis = []
    for i in string:
        if i in ('(', '['):
            parenthesis.append(i)
        elif i == ')':
            try:
                if parenthesis.pop() == '(':
                    continue
                else:
                    print("no")
                    break
            except:
                print("no")
                break
        elif i == ']':
            try:
                if parenthesis.pop() == '[':
                    continue
                else:
                    print("no")
                    break
            except:
                print("no")
                break
        elif i == '.':
            if parenthesis:
                print("no")
            else:
                print("yes")
